numpy/core/tests/test_multiarray.py

```diff
@@ -3783,7 +3784,7 @@ def test_roundtrip(self):
                         a, pickle.loads(pickle.dumps(a, protocol=proto)),
                         err_msg="%r" % a)
             del a, DATA, carray
-            gc.collect()
+            gc.collect(); gc.collect(); gc.collect()
```

Comment:
We need to garbage collect three times for pyoy? Can we add a comment to explain?

Reply:
Maybe out this is in a helper function, too?

Reply:
Short answer: PyPy requires multiple passes to resolve the confusing negative clauses in the note just before the [discussion in pep 442](https://www.python.org/dev/peps/pep-0442/#discussion). Unraveled, that note actually says "in some cases deallocators will still be called multiple times so deal with it", and PyPy deals with it by requiring multiple cycles.

Reply:
I will try to come up with a helper function with an appropriate docstring

Reply:
I seem to remember some code that iterates (with a for loop to avoid buggy infinite iteration), until gc.collect returns 0.

Reply:
@seberg perhaps you meant `test.support.gc_collect` from the stdlib. I will use that instead